[Doc] Introduce `<DatagridAG>` custom cell editors (#10410)
marmelab/react-admin: docs/DatagridAG.md

In addition to that, `<DatagridAG>` supports using [React Admin inputs](https://marmelab.com/react-admin/Inputs.html) as `cellEditor`, such as [`<TextInput>`](https://marmelab.com/react-admin/TextInput.html) or even [`<ReferenceInput>`](https://marmelab.com/react-admin/ReferenceInput.html).

This allows to leverage all the power of react-admin inputs in your grid, for example to edit a reference.

To use a React Admin input as `cellEditor`, you need to pass it as a *React Element*:

```tsx
import { List, ReferenceInput } from 'react-admin';
import { DatagridAG } from '@react-admin/ra-datagrid-ag';

export const CommentList = () => {
    const columnDefs = [
        // ...
        {
            field: 'post_id',
            cellEditor: (
                <ReferenceInput source="post_id" reference="posts" />
            ),
        },
    ];
    return (
        <List>
            <DatagridAG columnDefs={columnDefs} />
        </List>
    );
};
```

<video controls autoplay playsinline muted loop>

>   <source src="./assets/DatagridAG-ReferenceInput-AutocompleteInputAG.mp4" type="video/mp4"/>

Reviewer: ```suggestion
  <source src="https://react-admin-ee.marmelab.com/assets/DatagridAG-ReferenceInput-AutocompleteInputAG.mp4" type="video/mp4"/>
```

```
{% endraw %}

### `cellEditor`

In a column definition, you can use the `cellEditor` field to specify a custom cell editor. You can use any [Edit Component](https://www.ag-grid.com/react-data-grid/cell-editors/) supported by `ag-grid`, including [Custom Components](https://www.ag-grid.com/react-data-grid/cell-editors/#custom-components).

> In addition to that, `<DatagridAGClient>` supports using [React Admin inputs](https://marmelab.com/react-admin/Inputs.html) as `cellEditor`, such as [`<TextInput>`](https://marmelab.com/react-admin/TextInput.html) or even [`<ReferenceInput>`](https://marmelab.com/react-admin/ReferenceInput.html).

Reviewer: ```suggestion
In addition to that, `<DatagridAGClient>` supports using [React Admin inputs](./Inputs.md) as `cellEditor`, such as [`<TextInput>`](./TextInput.md) or even [`<ReferenceInput>`](./ReferenceInput.md).
```

This allows to leverage all the power of react-admin inputs in your grid, for example to edit a reference.

To use a React Admin input as `cellEditor`, you need to pass it as a *React Element*:

```tsx
import { List, ReferenceInput } from 'react-admin';
import { DatagridAGClient } from '@react-admin/ra-datagrid-ag';

export const CommentList = () => {
    const columnDefs = [
        // ...
        {
            field: 'post_id',
            cellEditor: (
                <ReferenceInput source="post_id" reference="posts" />
            ),
        },
    ];
    return (
        <List perPage={10000} pagination={false}>
            <DatagridAGClient columnDefs={columnDefs} />
        </List>
    );
};
```

<video controls autoplay playsinline muted loop>

>   <source src="https://react-admin-ee.marmelab.com /assets/DatagridAG-ReferenceInput-AutocompleteInputAG.mp4" type="video/mp4"/>

Author: ```suggestion
  <source src="https://react-admin-ee.marmelab.com/assets/DatagridAG-ReferenceInput-AutocompleteInputAG.mp4" type="video/mp4"/>
```

>   <source src="./assets/DatagridAG-ReferenceInput-AutocompleteInputAG.mp4" type="video/mp4"/>

Reviewer: ```suggestion
  <source src="https://react-admin-ee.marmelab.com /assets/DatagridAG-ReferenceInput-AutocompleteInputAG.mp4" type="video/mp4"/>
```
